packages/verified-fetch/src/index.ts

```diff
@@ -784,34 +792,75 @@ export interface VerifiedFetchInit extends RequestInit, ProgressOptions<BubbledP
   allowInsecure?: boolean
 }
 
+type Libp2pServiceType = Pick<DefaultLibp2pServices, 'dcutr' | 'identify' | 'keychain' | 'ping' > & Record<`delegatedRouting${number}`, any>
+
 /**
  * Create and return a Helia node
  */
 export async function createVerifiedFetch (init?: Helia | CreateVerifiedFetchInit, options?: CreateVerifiedFetchOptions): Promise<VerifiedFetch> {
+  let libp2p: Libp2p<Libp2pServiceType> | undefined
   if (!isHelia(init)) {
-    init = await createHeliaHTTP({
+    const libp2pConfig = libp2pDefaults()
+
+    libp2pConfig.start = false
+
+    libp2pConfig.transports = [webSockets(), webRTCDirect()]
+
+    libp2pConfig.peerDiscovery = [] // disable default bootstrap peers
+    libp2pConfig.addresses = {} // disable default listen addresses
+
+    // We only need client/listen/fetch based services
+    const fetchOnlyServices = {
+      dcutr: libp2pConfig.services.dcutr,
+      dht: kadDHT({
+        clientMode: true,
+        validators: {
+          ipns: ipnsValidator
+        },
+        selectors: {
+          ipns: ipnsSelector
+        }
+      }),
+      identify: libp2pConfig.services.identify,
+      identifyPush: libp2pConfig.services.identifyPush,
+      keychain: libp2pConfig.services.keychain,
+      ping: libp2pConfig.services.ping
+    }
+    // @ts-expect-error - borked serviceMap types
+    libp2pConfig.services = fetchOnlyServices
+    const routers = init?.routers ?? ['https://delegated-ipfs.dev']
+    for (let index = 0; index < routers.length; index++) {
+      const routerUrl = routers[index]
+      libp2pConfig.services[`delegatedRouting${index}`] = () => createDelegatedRoutingV1HttpApiClient(routerUrl)
+    }
+
+    libp2p = await createLibp2p(libp2pConfig)
+
+    init = await createHelia({
+      libp2p,
       blockBrokers: [
         trustlessGateway({
           allowInsecure: init?.allowInsecure,
           allowLocal: init?.allowLocal
-        })
+        }),
+        bitswap()
       ],
+      dns: createDns(init?.dnsResolvers),
```

Comment:
FYI: setting `libp2pConfig.dns` was causing some failures in tests. do we need that? dns should be handled at the helia layer shouldn't it?

cc @2color @achingbrain

Reply:
I think both Helia and libp2p rely on the DNS type. For libp2p it's mostly for resolving `dnsaddr` multiaddrs.